src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -448,6 +448,63 @@ public function isAbsolutePath($file)
     }
 
     /**
+     * Creates a temporary file with support for custom stream wrappers.
+     *
+     * @param string $dir    The directory where the temporary filename will be created.
+     * @param string $prefix The prefix of the generated temporary filename.
+     *                       Note: Windows uses only the first three characters of prefix.
+     *
+     * @return string The new temporary filename (with path), or false on failure.
+     */
+    public function tempnam($dir, $prefix)
+    {
+        $limit = 10;
+        list($scheme, $hierarchy) = $this->getSchemeAndHierarchy($dir);
+
+        // If no scheme or scheme is "file" create temp file in local filesystem
+        if (null === $scheme || 'file' === $scheme) {
+
+            $tmpFile = tempnam($hierarchy, $prefix);
+
+            // If tempnam failed or no scheme return the filename otherwise prepend the scheme
+
+            if (null !== $scheme) {
+                return $scheme.'://'.$tmpFile;
+            }
+
+            if (false !== $tmpFile) {
+                return $tmpFile;
+            }
+
+            throw new IOException('A temporary file could not be created');
+        }
+
+        // Loop until we create a valid temp file or have reached $limit attempts
+        for ($i = 0; $i < $limit; ++$i) {
+
+            // Create a unique filename
+            $tmpFile = $dir.'/'.$prefix.uniqid(mt_rand(), true);
+
+            // Use fopen instead of file_exists as some streams do not support stat
+            // Use mode 'x' to atomically check existence and create to avoid a TOCTOU vulnerability
+            $handle = @fopen($tmpFile, 'x');
+
+            // If unsuccessful restart the loop
+            if (false === $handle) {
+                continue;
+            }
+
+            // Close the file if it was successfully opened
+            @fclose($handle);
+
+            return $tmpFile;
+
+        }
+
+        return false;
```

Comment:
Shouldn't this throw instead of returning false? It would be more consistent with what's done a few lines above.